packages/itwin/quantity-formatting/src/components/quantityformat/FormatSelector.tsx

```diff
@@ -59,12 +62,18 @@ export const FormatSelector: React.FC<FormatSelectorProps> = ({
 
   const handleFormatSelect = React.useCallback(
     (key: string) => {
-      if (activeFormatSet?.formats) {
-        const formatDef = activeFormatSet.formats[key];
-        onListItemChange(formatDef, key);
+      const formatEntry = formatEntries.find(entry => entry.key === key);
+      if (formatEntry) {
+        onListItemChange(formatEntry.formatDef, key);
+      } else {
+        Logger.logWarning(`QuantityFormatting.FormatSelector`,`Format entry not found for key: ${key}`, {
```

Comment:
look at how we do logging in other pkgs, and make sure you set a logger namespace that others can use instead of hardcoding strings

Reply:
do we need an enum? isnt everything in this pkg going to run in frontend so the namespace isnt that helpful?